### DISCUSSION TITLE
Why does blankConnectionViewState expose limited options

### DISCUSSION THREAD

**Discussion by Alfonso-Martello** [iTwin/itwinjs-core#6096] [Q&A]
Hello, I have been working recently with blank viewer connections and have noticed that the [`blankConnectionViewState`](https://github.com/iTwin/viewer/blob/dad1ea4580ace8e56ec4a0d77e496fe6a0b52e2d/packages/modules/viewer-react/src/types.ts#L168C19-L168C19) prop only exposes limited options to manipulate the Viewer's default view. I am wondering why these options are limited?

Specifically:

- `BlankConnectionViewStateLookAt` provides limited options for looking at (eyePoint, targetPoint, and upVector are always required props). On the other hand, there are several flavors of "lookAt" methods supported by the platform (lookAt, lookAtVolume, lookAtGlobalLocation, etc). Even the [lookAt](https://www.itwinjs.org/reference/core-frontend/views/viewstate3d/lookat/) method supports several different types of arguments that are not supported for the blank connection.
- `BlankConnectionViewStateDisplayStyle` only allows setting the background color.
- `BlankConnectionViewStateVimapLayOptionsewFlags` only exposes 3 view flags.

**Comment by Alfonso-Martello** [comment]
Also including @Ellord207, who has also expressed interest in this topic.

**Comment by aruniverse** [comment]
The Viewer tries to choose reasonable defaults and exposes a subset of the vast and overwhelming iTwin.js APIs; like the majority of options we expose, they are requested by the user, validate if it makes sense, and then exposed if so. If you have a specific option you want exposed, please list it and an explanation why its needed.

**Reply by Ellord207** [reply]
While the iTwin.js platform does provide vast and overwhelming options to new users of the system, providing multiple, unexplained entry points to those options only exponentially increases the options.  When working with new developers to the platform, this is one of the largest issue I see, especially around the views.  Even mildly experienced platform developers do not always understanding the divide between the viewport, view state, and display styles.

While I agree with the intent behind selecting limited options, explaining that options and alternatives to the users (developers) is important.  I think it would majorly improve this feature to add a comment on the `BlankConnectionViewState` interface that it is "providing limited options for simplicity" and explain the alterative workflow (`viewportOptions: { viewState }`) to have more complete options.

**Reply by Ellord207** [reply]
Honestly, the complete lack of documentations around `BlankConnectionViewState` and it's properties interfaces is disappointing.

While many of the properties are fairly self explanatory, there is no hard connection between the viewer documentation and the core documentation although the properties themselves directly map between them.
